tests/testthat/test-parallel-tempering.R

```diff
@@ -261,3 +261,79 @@ test_that("tempering a model with no gradient produces one with no grad", {
   expect_false(bb$properties$has_gradient)
   expect_null(bb$gradient)
 })
+
+
+test_that("can use adaptive random walk with parallel tempering", {
+  ## We just do a short run here at present to make sure that things
+  ## work.  The model here is quite slow, mostly because of the prior;
+  ## something that we can probably improve later.  Proving that this
+  ## does the right thing statistically requires running for a long
+  ## time (a few minutes probably) and we should add some long running
+  ## tests later.
+  set.seed(1)
+  likelihood <- ex_mixture(5)
+  prior <- monty_dsl({
+    x ~ Normal(0, 10)
+  })
+  posterior <- likelihood + prior
+  sampler <- monty_sampler_parallel_tempering(
+    n_rungs = 10,
+    sampler = monty_sampler_adaptive(initial_vcv = matrix(0.1)))
+  res <- monty_sample(posterior, sampler, 7, n_chains = 2)
+  
+  expect_s3_class(res, "monty_samples")
+  expect_setequal(names(res$details$sampler),
+                  c("autocorrelation", "iteration", "mean", "scaling_history",
+                    "scaling_weight", "vcv", "weight"))
+  
+  ## Note that the hot chain never turns up here, as it comes from the
+  ## direct sample
+  ## 1 parameter x 1 parameter x 10 rungs x 2 chains
+  expect_equal(dim(res$details$sampler$autocorrelation), c(1, 1, 10, 2))
+  expect_equal(dim(res$details$sampler$vcv), c(1, 1, 10, 2))
```

Comment:
It will be nice to see how these end up looking over the rungs; something to visualise in the (eventual) PT vignette